test/selenium/testlib_avocado/seleniumlib.py

```diff
@@ -419,18 +419,10 @@ def add_authorised_ssh_key_to_user(self, pub_key=None, user=user):
         ssh_public_key = open("%s.pub" % pub_key).read()
         ssh_key_name = ssh_public_key.rsplit(" ", 1)[1]
 
-        self.click(self.wait_link('Accounts', cond=clickable))
-        self.wait_frame("users")
-        self.click(self.wait_xpath("//*[@class='cockpit-account-user-name']//a[contains(text(), '%s')]" % user, cond=clickable))
-
-        self.wait_id("account-page")
-        self.click(self.wait_id("authorized-key-add", cond=clickable))
-        self.send_keys(self.wait_id("authorized-keys-text", cond=visible), ssh_public_key)
-        self.click((self.wait_id("add-authorized-key", cond=clickable)))
-        self.wait_id("authorized-key-add", cond=clickable)
-        self.wait_xpath("//div[@class='comment' and contains(text(), '%s')]" % ssh_key_name)
-        self.wait_id("account-page")
-        self.mainframe()
+        old_ssh_user = self.machine.ssh_user
+        self.machine.ssh_user = "root"
```

Comment:
It's a chicken-egg thing: `self.machine.ssh_user` can't log in without having the key added to authorized_keys, and it needs to log in in order to add the key.

(Technically it can log in with a password, but I think `self.machine.execute` doesn't support that.)

I add a comment.

Reply:
Ah, of course -- thanks!